integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/DumpDefaultConfigs.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.tests
+
+import java.io.{BufferedOutputStream, DataOutputStream, FileOutputStream}
+import java.nio.charset.StandardCharsets
+import java.util.Locale
+
+import com.nvidia.spark.rapids.Arm.withResource
+import com.nvidia.spark.rapids.ShimReflectionUtils
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization.writePretty
+
+/**
+ * Dump all spark-rapids configs with their defaults.
+ */
+object DumpDefaultConfigs {
+
+  object Format extends Enumeration {
+    type Format = Value
+    val PLAIN, JSON = Value
+  }
+
+  def main(args: Array[String]): Unit = {
+    if (args.length < 2) {
+      System.err.println(s"Usage: ${this.getClass.getCanonicalName} {format} {output_path}")
+      System.exit(1)
+    }
+
+    val format = Format.withName(args(0).toUpperCase(Locale.US))
+    val outputPath = args(1)
+
+    println(s"Dumping all spark-rapids configs and their defaults at ${outputPath}")
+
+    // We use the reflection as RapidsConf should be accessed via the shim layer.
+    val clazz = ShimReflectionUtils.loadClass("com.nvidia.spark.rapids.RapidsConf")
```

Comment:
The logic for this class could be moved to RapidsConf which already generates files related to its settings.  This would be another file that is emitted when it is run, and integration_tests could pick up the file from this portion of the build.

Reply:
Hmm good suggestion. I think I added this class before imagining that this class would be useful only for integration test. Given that this class has been generalized enough (it takes the output format and the path as its params), I agree that `RapidsConf` is a better place to put this logic. I will make this change soon.